lib/rack/request.rb

```diff
@@ -19,6 +19,20 @@ class << self
 
     self.ip_filter = lambda { |ip| ip =~ /\A127\.0\.0\.1\Z|\A(10|172\.(1[6-9]|2[0-9]|30|31)|192\.168)\.|\A::1\Z|\Afd[0-9a-f]{2}:.+|\Alocalhost\Z|\Aunix\Z|\Aunix:/i }
     SCHEME_WHITELIST = %w(https http).freeze
+    if Object.respond_to?(:deprecate_constant)
+      deprecate_constant :SCHEME_WHITELIST
+    else
+      if $VERBOSE
+        warn("SCHEME_WHITELIST is deprecated and will be removed in a future version of Rack. Please use Rack::Request.scheme_allowlist instead.")
+      end
+    end
+
+    def self.scheme_allowlist
```

Comment:
I think we don't need a class method for this. This constant was never made to be exposed as public. Just a public `ALLOWED_SCHEMAS` is fine.